torch/csrc/jit/python/pybind_utils.h

```diff
@@ -163,15 +194,19 @@ struct VISIBILITY_HIDDEN PythonFutureWrapper
                 e.what()));
             {
               pybind11::gil_scoped_acquire ag;
+              py::object exceptionType = e.type();
```

Comment:
instead of catching the python error, acquiring gil and creating another python error, can we just rethrow the python error here, or modify the `python_error` error message then rethrowing? I feel like all these piece of code is not necessary. i.e. can we do something like this? https://github.com/pytorch/pytorch/blob/aa2fede201078ee91529edcccb734e5e59249ac2/torch/csrc/autograd/python_engine.cpp#L115